app/site/controllers.py

```diff
@@ -128,9 +128,15 @@ def publisher_dashboard(publisher):
 
 @site_blueprint.route("/search", methods=["GET"])
 def search_package():
-    q = request.args.get('q')
-    datapackage_list = DataPackageQuery(query_string=q).get_data()
-    return render_template("search.html",
-                           datapackage_list=datapackage_list,
-                           total_count=len(datapackage_list),
-                           query_term=q), 200
+    try:
+        q = request.args.get('q')
+        if not q.strip():
```

Comment:
I'm not sure you need the if statement here around the whole render method. The if really only applies to getting the datapackage_list.

In addition, i think the search page should probably show a default set of packages if no query (ie. even if you don't search, but go to page directly you get results ...)

Code could be:

```
q = request.args.get('q').strip()
if q:
   datapackage_list = ...
else:
   datapackage_list = search with empty query - show me everything (limited to top x items)

return render_template(...)
```